api/src/main/java/org/openmrs/module/eptsreports/reporting/library/queries/QualityImprovement2020Queries.java

```diff
@@ -2741,4 +2741,164 @@ public static String getPatientsWithRestartedStateOfStayAndCd4RequestQuery() {
         + "  AND e.location_id = :location "
         + "GROUP BY pa.patient_id";
   }
+
+  /**
+   * todos os utentes com registo de “Tem sintomas?” (TB) = “Sim” em uma consulta clínica (Ficha
+   * Clínica) ocorrida durante o período de revisão;
+   *
+   * @return {@link String}
+   */
+  public static String getPatientsWithSintomasTBSim() {
+    return "SELECT p.patient_id, "
+        + "               Min(e.encounter_datetime) AS data_presuntivo_tb "
+        + "        FROM   patient p "
+        + "                   INNER JOIN encounter e "
+        + "                              ON e.patient_id = p.patient_id "
+        + "                   INNER JOIN obs o "
+        + "                              ON o.encounter_id = e.encounter_id "
+        + "        WHERE  p.voided = 0 "
+        + "          AND e.voided = 0 "
+        + "          AND o.voided = 0 "
+        + "          AND e.location_id = :location "
+        + "          AND e.encounter_datetime >= :startDate "
+        + "          AND e.encounter_datetime <= :endDate "
+        + "          AND e.encounter_type = ${6} "
+        + "          AND o.concept_id = ${23758} "
+        + "          AND o.value_coded = ${1065} "
+        + "        GROUP  BY p.patient_id";
+  }
```

Comment:
Can you create one function that receives 2 params, a concept id and list of concepts as value coded a will produce your patient id and min(encounter_datetime)

Reply:
Done